weave/tutorial-eval.mdx

```diff
@@ -1,259 +1,447 @@
-  }
-);
+import OpenAI from 'openai';
 ```
-</CodeGroup>
+</Tab>
+</Tabs>
 
-<Note>
-To make your own scoring function, learn more in the [Scorers](/weave/guides/evaluation/scorers) guide.
 
-In some applications we want to create custom `Scorer` classes - where for example a standardized `LLMJudge` class should be created with specific parameters (e.g. chat model, prompt), specific scoring of each row, and specific calculation of an aggregate score. See the tutorial on defining a `Scorer` class in the next chapter on [Model-Based Evaluation of RAG applications](/weave/tutorial-rag#optional-defining-a-scorer-class) for more information.
-</Note>
+## Build a `Model`
 
-## 4. Run the evaluation
+In Weave, [`Models` are objects](/weave/guides/core-types/models) that capture both the behavior of your model/agent (logic, prompt, parameters) and its versioned metadata (parameters, code, micro-config) so you can track, compare, evaluate and iterate reliably.
 
-Now, you're ready to run an evaluation of `ExtractFruitsModel` on the `fruits` dataset using your scoring function.
+When you instantiate a `Model`, Weave automatically captures its configuration and behaviors and updates the version when there are changes. This allows you to track its performance over time as you iterate on it.
 
-<CodeGroup>
-```python lines
-import asyncio
-import weave
-from weave.scorers import MultiTaskBinaryClassificationF1
-
-weave.init('intro-example')
-
-evaluation = weave.Evaluation(
-    name='fruit_eval',
-    dataset=dataset, 
-    scorers=[
-        MultiTaskBinaryClassificationF1(class_names=["fruit", "color", "flavor"]), 
-        fruit_name_score
-    ],
-)
-print(asyncio.run(evaluation.evaluate(model)))
-# if you're in a Jupyter Notebook, run:
-# await evaluation.evaluate(model)
-```
+`Model`s are declared by subclassing `Model` and implementing a `predict` function definition, which takes one example and returns the response.
 
-```typescript TypeScript
-import * as weave from 'weave';
+The following example model uses OpenAI to extract the names, colors, and flavors of alien fruits from sentences sent to it.
+
+<Tabs>
+  <Tab title="Python">
+  ```python lines {5,9}
+  class ExtractFruitsModel(weave.Model):
+      model_name: str
+      prompt_template: str
+
+      @weave.op()
+      async def predict(self, sentence: str) -> dict:
+          client = openai.AsyncClient()
+
+          response = await client.chat.completions.create(
+              model=self.model_name,
+              messages=[
+                  {"role": "user", "content": self.prompt_template.format(sentence=sentence)}
+              ],
+          )
+          result = response.choices[0].message.content
+          if result is None:
+              raise ValueError("No response from model")
+          parsed = json.loads(result)
+          return parsed
+  ```
+  </Tab>
+  <Tab title="TypeScript">
+  ```typescript
+  // Note: weave.Model is not supported in TypeScript yet.
```

Comment:
Since this is a work around, I wonder if it makes sense to also move this code comment as regular text above the code snippet. It feels less "sneaky". It was a little jarring, tbh, to click TypeScript (expecting support for it) only to see a word around and acknowledging said work around as a comment.